dev-lang/swi-prolog/swi-prolog-8.5.20.ebuild

```diff
@@ -0,0 +1,157 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake desktop flag-o-matic java-pkg-opt-2 toolchain-funcs xdg
+
+PATCHSET_VER="0"
+
+MY_P="swipl-${PV}"
+DESCRIPTION="Versatile implementation of the Prolog programming language"
+HOMEPAGE="https://www.swi-prolog.org/"
+SRC_URI="https://www.swi-prolog.org/download/devel/src/${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+IUSE="X archive berkdb +cli debug doc +gmp gui +ipc java minimal odbc pcre pgo qt6 ssl test yaml"
+RESTRICT="!test? ( test )"
+
+# See cmake/PackageSelection.cmake and cmake/DocDepends.cmake
+REQUIRED_USE="
+	doc? ( archive )
+	minimal? ( !archive !doc !ipc !ssl !test )
+"
+
+RDEPEND="
+	!minimal? ( dev-libs/ossp-uuid )
+	X? (
+		media-libs/freetype:2
+		media-libs/fontconfig:1.0
+		media-libs/libjpeg-turbo:=
+		x11-libs/libICE
+		x11-libs/libSM
+		x11-libs/libX11
+		x11-libs/libXft
+		x11-libs/libXinerama
+		x11-libs/libXpm
+		x11-libs/libXt
+	)
+	sys-libs/ncurses:=
+	sys-libs/zlib:=
+	virtual/libcrypt:=
+	archive? ( app-arch/libarchive:= )
+	berkdb? ( >=sys-libs/db-4:= )
+	cli? (
+		dev-libs/libedit
+		sys-libs/readline:=
+	)
+	gmp? ( dev-libs/gmp:0= )
+	gui? (
+		!qt6? (
+			dev-qt/qtgui:5
+			dev-qt/qtwidgets:5
+		)
+		qt6? ( dev-qt/qtbase:6[gui(+),widgets(+)] )
+	)
+	java? ( >=virtual/jdk-1.8:* )
```

Comment:
Why not put `virtual/jre` here and `virtual/jdk`  in DEPEND as is described in https://wiki.gentoo.org/wiki/Java_Developer_Guide#Using_java-pkg-opt-2.eclass?

Reply:
Applied, however I don't know if that's right because swipl uses JNI